contracts/MyNFT.sol

```diff
@@ -0,0 +1,123 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.7;
+
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+import "@openzeppelin/contracts/utils/Strings.sol";
+import "./interfaces/IMyNFT.sol";
+
+/**
+ * @title Nft
+ * @author Pavel Zanozin
+ * @notice This SC allows users buy NFT and register
+ * ownership on blockchain
+ */
+contract MyNFT is ERC721, Ownable, IMyNFT {
+    using Counters for Counters.Counter;
+    using Strings for uint256;
+
+    /**
+     * @dev contain how many tokens was minted at
+     * current time.
+     *
+     * @notice using 'Counter' library from OpenZeppelin.
+     */
+    Counters.Counter public tokenCountId;
+
+    /**
+     * @dev contain NFT price.
+     *
+     * @notice can be changed by owner.
+     */
+    uint256 public nftPrice;
+
+    /**
+     * @dev contain max tokens supply
+     *
+     */
+    uint256 public tokenSupply = 5;
+
+    /**
+     * @dev contain baseURI.
+     *
+     * @notice can be changed by owner.
+     */
+    string public baseURILink;
+
+    /**
+     * @dev Input uint256 for 'price_' and string for 'baseURI_'
+     *
+     * @param price_ - price per nft
+     * @param baseURI_ - base URI fro nft's
+     */
+    constructor(uint256 price_, string memory baseURI_) ERC721("MyNft", "MNT") {
+        setNftPrice(price_);
+        setBaseURI(baseURI_);
+    }
+
+    /**
+     * @dev allows user buy nft's for the price which set
+     * by the owner . Don't take any params
+     *
+     * @notice can be by call by any one . revert transaction if
+     * all nft's was minted or paymant value too low
+     */
+    function buy() external payable override {
+        require(msg.value == nftPrice, "Error : payment value too low");
+        require(
+            tokenCountId.current() <= tokenSupply,
```

Comment:
for gas optimization create variable for tokenCountId.current() for interact with memory and not interact with storage every time